src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1096,5 +1102,41 @@ private static void ThrowMinMaxException<T>(T min, T max)
         {
             throw new ArgumentException(SR.Format(SR.Argument_MinMaxValue, min, max));
         }
+
+        public static double ScaleB(double x, int n)
```

Comment:
It might be good to add a small comment explaining how this function works.

Just something similar to or roughly detailing that `scalbn(x, n)` is `x * 2^n`. So we construct a `double` from `2^n` by building the correct biased exponent (https://github.com/dotnet/runtime/pull/42476/files#diff-29944976f2a0c621a8f96a2e75ccdf55R1138). 

If `n` is greater than the maximum exponent (`1023` for double, `127` for float) or smaller then the minimum exponent (`-1022` for double, `-126` for float), then we need to make an adjustment to compute the correct result.